### PR TITLE
Fix tls handshaking issue

### PR DESCRIPTION
**Pull request by PabasaraDilshan**
## Description:

<details>
  <summary>Exception log</summary>

   
```
00:00:00.176 WIF: Checking connection...
00:00:00.177 WIF: Attempting connection...
00:00:01.001 WIF: Connecting to AP1 XXXX Channel 6 BSSId 82:D7:F4:D3:70:AC in mode 11n as XXXX-XXXX-XXX-00005-6237...
00:00:02.560 WIF: Checking connection...
00:00:02.561 WIF: Attempting connection...
00:00:03.509 WIF: Checking connection...
00:00:03.510 WIF: Attempting connection...
00:00:04.511 WIF: Checking connection...
00:00:04.511 WIF: Attempting connection...
00:00:05.514 WIF: Checking connection...
00:00:05.515 WIF: Attempting connection...
00:00:06.307 QPC: Reset
00:00:06.520 WIF: Checking connection...
00:00:06.520 WIF: Attempting connection...
00:00:07.524 WIF: Checking connection...
00:00:07.524 WIF: Connected
00:00:07.776 HTP: Web server active on XXX-XXXX-XXX-00005-6237 with IP address 192.168.9.138
00:00:08.279 APP: Boot Count 20
00:00:08.280 WIF: Sending Gratuitous ARP
00:00:08.281 NTP: Sync time...
00:00:08.513 WIF: DNS resolved '2.pool.ntp.org' (162.159.200.1) in 231 ms
00:00:09.518 NTP: No reply from 162.159.200.1
00:00:09.521 NTP: Sync time...
00:00:09.560 WIF: DNS resolved '2.europe.pool.ntp.org' (45.87.77.15) in 38 ms
00:00:10.566 NTP: No reply from 45.87.77.15
00:00:10.569 MQT: Attempting connection...
00:00:10.663 WIF: DNS resolved 'xxxxxxxxxxxx.azure-devices.net' (20.44.8.226) in 93 ms
00:00:10.664 MQT: Authenticating with an Azure IoT Hub Token
00:00:10.669 MQT: sha256 dataToSign is 'xxxxxxxxxxxx.azure-devices.net%2Fdevices%2FXXXX-XXXX-XXXX-00005
1707944171'
00:00:10.681 MQT: sha256 signature is 'xxxxxxxxxxxxxxxxxxxx='
00:00:10.688 MQT: Azure IoT Hub SAS Token is 'SharedAccessSignature sr=xxxxxxxxxxxxxxx.azure-devices.net%2Fdevices%2FXXXX-XXXXX-XXXX-00005&sig=xxxxxxxxxxxxxxx&se=1707944171'
00:00:10.707 MQT: Configure TLS fingerprint
00:00:10.711 MQT: Mqtt Client Connecting

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (28):
epc1=0x40202376 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

>>>stack>>>

ctx: sys
sp: 3fffed50 end: 3fffffb0 offset: 0190
3fffeee0:  098b70a4 3ffee9a0 3fff0174 40202374  
3fffeef0:  00000000 02fb5ad2 3ffed0f8 60000600  
3fffef00:  098b6111 3ffee9a0 3fff0100 40257ba6  
3fffef10:  40286298 3fff0174 3ffee9a0 40258b12  
3fffef20:  60000600 3fff0174 3ffee9a0 402862a5  
3fffef30:  402862ea 3fffdab0 00000000 3fffdcb0  
3fffef40:  3ffee9b0 3fffdad0 3fff1d08 40254c1c  
3fffef50:  40000f49 40000f49 3fffdab0 40000f49  
3fffef60:  40000e19 00000005 000b034c 00000000  
3fffef70:  00000000 aa55aa55 000000ed 40105b65  
3fffef80:  40105b6b 000b034c 00000000 387eeddc  
3fffef90:  4010000d 9fd13802 8f502524 44df11cf  
3fffefa0:  4027ed18 3fffef3c 4027ecc9 3ffffaa8  
3fffefb0:  3fffffc0 00000000 00000000 feefeffe  
3fffefc0:  feefeffe feefeffe feefeffe feefeffe  
3fffefd0:  feefeffe feefeffe feefeffe feefeffe  
3fffefe0:  feefeffe feefeffe feefeffe feefeffe  
3fffeff0:  feefeffe feefeffe feefeffe feefeffe  
3ffff000:  feefeffe feefeffe feefeffe feefeffe  
3ffff010:  feefeffe feefeffe feefeffe feefeffe  
3ffff020:  feefeffe feefeffe feefeffe feefeffe  
3ffff030:  feefeffe feefeffe feefeffe feefeffe  
3ffff040:  feefeffe feefeffe feefeffe feefeffe  
3ffff050:  feefeffe feefeffe feefeffe feefeffe  
3ffff060:  feefeffe feefeffe feefeffe feefeffe  
3ffff070:  feefeffe feefeffe feefeffe feefeffe  
3ffff080:  feefeffe feefeffe feefeffe feefeffe  
3ffff090:  feefeffe feefeffe feefeffe feefeffe  
3ffff0a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff100:  feefeffe feefeffe feefeffe feefeffe  
3ffff110:  feefeffe feefeffe feefeffe feefeffe  
3ffff120:  feefeffe feefeffe feefeffe feefeffe  
3ffff130:  feefeffe feefeffe feefeffe feefeffe  
3ffff140:  feefeffe feefeffe feefeffe feefeffe  
3ffff150:  feefeffe feefeffe feefeffe feefeffe  
3ffff160:  feefeffe feefeffe feefeffe feefeffe  
3ffff170:  feefeffe feefeffe feefeffe feefeffe  
3ffff180:  feefeffe feefeffe feefeffe feefeffe  
3ffff190:  feefeffe feefeffe feefeffe feefeffe  
3ffff1a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff200:  feefeffe feefeffe feefeffe feefeffe  
3ffff210:  feefeffe feefeffe feefeffe feefeffe  
3ffff220:  feefeffe feefeffe feefeffe feefeffe  
3ffff230:  feefeffe feefeffe feefeffe feefeffe  
3ffff240:  feefeffe feefeffe feefeffe feefeffe  
3ffff250:  feefeffe feefeffe feefeffe feefeffe  
3ffff260:  feefeffe feefeffe feefeffe feefeffe  
3ffff270:  feefeffe feefeffe feefeffe feefeffe  
3ffff280:  feefeffe feefeffe feefeffe feefeffe  
3ffff290:  feefeffe feefeffe feefeffe feefeffe  
3ffff2a0:  559cdc08 28cb484a 73891198 87825557  
3ffff2b0:  23f8b704 963d018e a59a0a80 c8cbf2aa  
3ffff2c0:  5c5c5c5c 5c5c5c5c 5c5c5c5c 5c5c5c5c  
3ffff2d0:  5c5c5c5c 5c5c5c5c 5c5c5c5c 5c5c5c5c  
3ffff2e0:  9b0660c1 a5ca844a f6ee5d4f 185e1620  
3ffff2f0:  b511515a 53c3d2ae 34c40adc 9b0c8afc  
3ffff300:  2a83b2c2 cb75fb94 c812879d f81bf384  
3ffff310:  6753acd8 bcd1714e 3f7bf724 25e872e4  
3ffff320:  0cd60ff7 bb68e2e4 7bc41c44 e45079a6  
3ffff330:  a20b6f45 8364890d fdf813e8 eb203f8c  
3ffff340:  3fc01b45 e2bc40cb a53b40f2 997e4ff8  
3ffff350:  2242051f 8f496bc1 1f24bba7 24af1a0c  
3ffff360:  71adeabd 83f3a01a 06d35d8d 554f8a51  
3ffff370:  80000000 00000000 00000000 00000000  
3ffff380:  00000000 00000000 00000000 00000300  
3ffff390:  ea49e19c f45c858b c64bea1a 05726879  
3ffff3a0:  c7acc792 cadb2b65 4c756bac 418676e9  
3ffff3b0:  8d0d3879 bc05d388 40831ce2 1b471088  
3ffff3c0:  b8d873fb b6c94abb 79f314ba 762ef0ca  
3ffff3d0:  b72598d3 b304fbe0 c9ac4735 ad89dbe6  
3ffff3e0:  41dbeacb 277075e1 91daacf3 d9975943  
3ffff3f0:  b5f09fad d91c0271 2197740a daf7dcae  
3ffff400:  67ef7170 1d7b1e42 d0355146 f0cbba2e  
3ffff410:  7d0014d6 e276a67f 5ba658b1 014bf92c  
3ffff420:  22881fc9 d97d9e4a d98cb232 3db076c5  
3ffff430:  5fde6914 37eaf8a1 ce636355 51376d64  
3ffff440:  5f65e889 160a3461 049b8132 53d2ec7e  
3ffff450:  00000100 1a0b47a0 2151dd47 263c45e0  
3ffff460:  e3c017f3 aeaf69c5 d1accb8c ec18751f  
3ffff470:  9bc44386 00000000 00000000 00000000  
3ffff480:  40103a20 00080000 6c302073 00000020  
3ffff490:  3ffff610 00000008 3ffff540 402015c7  
3ffff4a0:  1f054222 c16b498f a7bb241f 0c1aaf24  
3ffff4b0:  bdeaad71 1aa0f383 8d5dd306 518a4f55  
3ffff4c0:  00000080 00000000 00000000 00000000  
3ffff4d0:  00000000 00000000 00000000 00030000  
3ffff4e0:  02dd0292 598927eb df446094 9b48a04b  
3ffff4f0:  917e3d8a 6275c70d cc6d4df6 d291fa0a  
3ffff500:  3ffff610 3ffff540 00000020 402014d3  
3ffff510:  00000000 00000020 dfecc90d 3fff4b54  
3ffff520:  000000d0 3ffff680 4029b224 402015f4  
3ffff530:  000000d0 3ffff680 4029b224 402017f1  
3ffff540:  4029b224 1f054222 c16b498f a7bb241f  
3ffff550:  0c1aaf24 bdeaad71 1aa0f383 8d5dd306  
3ffff560:  518a4f55 04b7f823 8e013d96 800a9aa5  
3ffff570:  aaf2cbc8 5c5c5c5c 5c5c5c5c 5c5c5c5c  
3ffff580:  5c5c5c5c 5c5c5c5c 00000060 00000000  
3ffff590:  1a0b47a0 2151dd47 263c45e0 e3c017f3  
3ffff5a0:  aeaf69c5 d1accb8c ec18751f 9bc44386  
3ffff5b0:  aeaf69c5 d1accb8c ec18751f 9bc44386  
3ffff5c0:  696e692d 696f7468 75622d6f 76657273  
3ffff5d0:  69676874 2d63656e 7472616c 2d75732e  
3ffff5e0:  617a7572 652d6465 76696365 732e6e65  
3ffff5f0:  74000100 0102000d 00040002 0401000a  
3ffff600:  4f8238d2 dafb773c f8ade3c6 ff97f5f9  
3ffff610:  b8bc11fb 80214127 089fb5a1 7b40065e  
3ffff620:  08531e6e fece0e24 c214f91d 00fbf0bf  
3ffff630:  71e022de 93ea1c4d e4f73825 7d3b7012  
3ffff640:  88a568d8 5d3a09d3 2472ff52 9b54b8c2  
3ffff650:  7c3e02fe dee2d132 f92441e2 50ef4334  
3ffff660:  e42df552 8aab6d84 23d52195 12f2cafe  
3ffff670:  93fda833 6291460d 8e7b47f0 e77ce492  
3ffff680:  0de080e6 8f044ed8 403f2278 51a71bbb  
3ffff690:  1e5f9917 a6cb4339 d2c14451 ecbb75d1  
3ffff6a0:  c0e9c154 05acbb97 9af137df 1b5f180b  
3ffff6b0:  3f0fc0c0 c584814f 93b475db 66c4739d  
3ffff6c0:  00000100 299b4f0f 5c3d77dd ccbbd8b7  
3ffff6d0:  334697b1 86197868 4a956ba0 39d7875d  
3ffff6e0:  793ca3b0 fa87303a 36bbf905 f5438fbe  
3ffff6f0:  75ab05fc 3ffff770 0000000c 3ffff744  
3ffff700:  3fff51ec 3ffff740 00000000 402014ea  
3ffff710:  00000000 00000040 3ffff7c0 402015c7  
3ffff720:  e5974708 d19b9ca1 1b6fcac2 3fff52e8  
3ffff730:  4029b224 00000004 3fff516c 4026d579  
3ffff740:  00000080 2d696e69 68746f69 6f2d6275  
3ffff750:  73726576 74686769 6e65632d 6c617274  
3ffff760:  2e73752d 72757a61 65642d65 65636976  
3ffff770:  656e2e73 00010074 0d000201 02000400  
3ffff780:  0a000104 00000000 00000080 00000000  
3ffff790:  45e9a1cc e7db8e89 023ae67d 91c93a8d  
3ffff7a0:  3bd7295f aba78746 8776699b ca4ebabf  
3ffff7b0:  3a3087fa 05f9bb36 be8f43f5 5be0cd19  
3ffff7c0:  4029b224 e5974708 d19b9ca1 1b6fcac2  
3ffff7d0:  e8feef38 2db40fe1 bcbf7b86 a0470b1a  
3ffff7e0:  47dd5121 e0453c26 f317c0e3 c569afae  
3ffff7f0:  00000000 00000000 0000001f 40100b38  
3ffff800:  5c5c5c5c 5c5c5c5c 3fffc228 40106305  
3ffff810:  4000050c 3ffff861 3fff52dc 3fff5238  
3ffff820:  40266328 00000030 00000010 ffffffff  
3ffff830:  00000000 00000000 0000001f 40100b38  
3ffff840:  3fff88a6 00005d98 3fffc228 40106305  
3ffff850:  00000000 00000000 0000001f 40100b38  
3ffff860:  00000000 00000000 0000001f 40100b38  
3ffff870:  4000050c 3ffedb2c 3fffc228 40106305  
3ffff880:  4000050c 00000030 00000019 fffffffe  
3ffff890:  401039de 00000030 0000001f fffffffe  
3ffff8a0:  3ffe9f1c 3ffe9f1c 00000001 0673c3a5  
3ffff8b0:  00000005 00000000 00000020 40100b38  
3ffff8c0:  00000005 00000000 00000020 40100b38  
3ffff8d0:  00000005 00000000 00000020 40100b38  
3ffff8e0:  00000020 40100b38 00000005 401027ac  
3ffff8f0:  3ffe9695 40105a5f 3ffed0f8 40105a5f  
3ffff900:  00000005 00000000 00000020 40100b38  
3ffff910:  00000005 00000000 00000005 401027ac  
3ffff920:  00000005 00000000 00000020 40100b38  
3ffff930:  40103343 3ffed0f8 00000005 401027ac  
3ffff940:  3ffe9695 40105a5f 3ffed0f8 40103524  
3ffff950:  40103343 3ffed0f8 00000000 40100b38  
3ffff960:  00000008 098a62e3 3ffedb2c 40103524  
3ffff970:  3ffe9f34 00000000 00000000 00000001  
3ffff980:  00000008 098a62e3 401039e6 00000100  
3ffff990:  00000000 00000000 0000001f 40100b38  
3ffff9a0:  00000000 00000000 3fffc228 40106305  
3ffff9b0:  4000050c 00000000 0000001f 40100b38  
3ffff9c0:  40269993 00000030 00000000 ffffffff  
3ffff9d0:  4024b43f 3fff465c 0000134c ffc85adb  
3ffff9e0:  00000005 00000545 00000000 fffffffe  
3ffff9f0:  ffffffff 3fffc6fc 00000001 3fff45a4  
3ffffa00:  02d1342d 00000008 00000001 00000030  
3ffffa10:  ffffffff 3fffc6fc 00000001 3fff45a4  
3ffffa20:  02d128b5 00000008 00000001 00000030  
3ffffa30:  3fff2584 00000000 00000000 402660f3  
3ffffa40:  00004428 00000885 00000885 401019d3  
3ffffa50:  3fff2584 3fff8864 00000000 3fff5934  
3ffffa60:  00000001 00000000 00000020 40101c5b  
3ffffa70:  007a1200 ffc8399a 4bc6a700 00000000  
3ffffa80:  007a1200 ffc85aee 00000000 40100b38  
3ffffa90:  00000000 00000000 00000001 40100b38  
3ffffaa0:  00000008 02d133f5 3fff45a4 02d13439  
3ffffab0:  00000008 00000001 40254d37 3fffefa0  
3ffffac0:  00000000 00000000 3fff465c 40254d7c  
3ffffad0:  00000008 02d13438 3fff45a4 4024b43a  
3ffffae0:  00000091 00000005 a1b8083f 00000000  
3ffffaf0:  40269f00 40269eac 3fff465c 40269a8a  
3ffffb00:  3fff4b5c 3fff693c 3fff465c 4029b224  
3ffffb10:  3fff45a4 00000001 3fff45a4 4024b558  
3ffffb20:  3fff45a4 3fff83e4 00000000 4024b70d  
3ffffb30:  3fff693c 00000004 3fff1bd8 40252a28  
3ffffb40:  3fff45a4 3ffffb90 3fff1bd8 3fff1bd8  
3ffffb50:  3fff45a4 3ffffb90 000022b3 4024b884  
3ffffb60:  402591d8 e2082c14 00ff1bd8 40252a28  
3ffffb70:  3ffefcf4 3ffffbf0 3fff1bd8 3fff45a4  
3ffffb80:  3ffefcf4 3ffffbf0 3fff1bd8 4025077a  
3ffffb90:  402591d8 e2082c14 3fff0338 40203874  
3ffffba0:  3fff6ad4 3fff6c6c 3fff06b6 00000001  
3ffffbb0:  00000001 00000001 4024b7e4 00000008  
3ffffbc0:  0000006b 0000086b 3ffffcc8 00000000  
3ffffbd0:  3ffe8678 00000000 3fff01b8 402369c1  
3ffffbe0:  00000001 3fff0338 00000001 4025b89a  
3ffffbf0:  4f485441 4d532d4d 2d545241 47554c50  
3ffffc00:  3030302d 4c2f3530 00005457 3fff69d2  
3ffffc10:  00000000 3fff69d1 3ffffcc0 3fff02bc  
3ffffc20:  00000000 3ffe8304 00000015 4025b039  
3ffffc30:  3ffffcad 00000000 4025000a 40256ead  
3ffffc40:  3ffffcad 00000008 3fff01bc 4025b07c  
3ffffc50:  3ffffd00 3ffffcd0 00000008 0000002d  
3ffffc60:  3fff17b4 00000008 3ffffc80 40252410  
3ffffc70:  00000002 00000009 00000000 40252888  
3ffffc80:  303a3030 30313a30 3fff6a00 004b004f  
3ffffc90:  004e203a 70657220 6620796c 206d6f72  
3ffffca0:  382e3534 37372e37 0d35312e 4025000a  
3ffffcb0:  3ffffd00 3ffffcf0 0000000c 4025b039  
3ffffcc0:  3ffffd00 3ffffcd0 3fff6900 0034003f  
3ffffcd0:  00000000 3fff6ad4 004b004f 80101350  
3ffffce0:  3fff6c6c 00b300bf 80257bdc 402591d8  
3ffffcf0:  e2082c14 00000000 00000000 4021787a  
3ffffd00:  00002c9f 3fff6ad4 3fff6c6c 3fff341c  
3ffffd10:  3fff42eb 3fff42ff 3ffefcf4 40258a39  
3ffffd20:  3fff01bc 3fff01b8 3fff02bc 3ffffd50  
3ffffd30:  3fff01bc 3fff01b8 3fff02bc 40236b70  
3ffffd40:  3fff01bc 3fff01b8 00000002 40243942  
3ffffd50:  3ffffd70 3ffffd60 0000000c 3fff1b6c  
3ffffd60:  00000001 3fff6a24 00000020 40101c5b  
3ffffd70:  00000057 0000004d 3fff69bc 40250c2c  
3ffffd80:  00000000 00000003 00000001 40250ff1  
3ffffd90:  0000000f 3fff6afc 3fff698c 40265a02  
3ffffda0:  3ffffe40 3ffffe30 00000008 40266000  
3ffffdb0:  3ffffe40 3ffffe30 0000000c 3fff69bc  
3ffffdc0:  00004300 00000860 00000860 401019d3  
3ffffdd0:  00004398 00000873 00000873 00002c39  
3ffffde0:  000042b0 00000856 00000856 401019d3  
3ffffdf0:  00000000 00000003 00000020 00002c39  
3ffffe00:  00000000 00000000 00000020 40101c5b  
3ffffe10:  00000000 00000003 3fff6a24 4025e949  
3ffffe20:  3ffffe40 3ffffe30 3fff693c 4025e091  
3ffffe30:  00000000 3fff2060 3fff6a0c 40248c0a  
3ffffe40:  00000000 00000001 3ffffed0 40248c31  
3ffffe50:  00000000 00000001 00000000 40229838  
3ffffe60:  2e323931 2e383631 00312e39 3fff01b8  
3ffffe70:  00000000 00000000 3b9aca00 00000000  
3ffffe80:  6f702e30 6e2e6c6f 6f2e7074 3f006772  
3ffffe90:  00000000 30ff2584 30303a30 40248602  
3ffffea0:  00000000 ee6b2800 00002800 34314e31  
3ffffeb0:  00000000 00000001 3ffffed0 00000000  
3ffffec0:  00000016 00000008 3fff45a4 4024b0d7  
3ffffed0:  40258f48 00000008 3ffefcf4 40258a39  
3ffffee0:  00000000 00000008 3ffefcf4 4025049a  
3ffffef0:  00000000 00000000 4bc6a7f0 00000000  
3fffff00:  00000000 4bc6a7f0 401012c2 ef9db22d  
3fffff10:  00000000 4bc6a7f0 00002c3f 3fff1d08  
3fffff20:  00000000 00000000 4bc6a7f0 00000000  
3fffff30:  00000016 00000009 401012c2 6083126e  
3fffff40:  00000000 3fff341c 00000032 3fff1d08  
3fffff50:  00000224 3ffef01c 000000fa 3fff1d08  
3fffff60:  3fffdad0 00000000 3ffef01c 40243c6a  
3fffff70:  3fffdad0 00000000 00002c3e 40243cc8  
3fffff80:  00000000 00000000 00000001 40100b38  
3fffff90:  3fffdad0 00000000 3fff1cc8 3fff1d08  
3fffffa0:  3fffdad0 00000000 3fff1cc8 40254dc0  
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

 ets Jan  8 2013,rst cause:1, boot mode:(3,6)

load 0x4010f000, len 3584, room 16 
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.001 HDW: ESP8285H16
00:00:00.054 CFG: Loaded from flash at F4, Count 32
00:00:00.059 QPC: Count 1
00:00:00.061 CFG: CR 470/699, Busy 0
00:00:00.065 TYA: Active=0
00:00:00.070 ROT: Mode 1
00:00:00.072 SRC: Restart
00:00:00.073 Project tasmota - Tasmota Version 13.3.0.5(tasmota)-2_7_6(2024-02-14T02:06:19)
00:00:00.176 WIF: Checking connection...
00:00:00.177 WIF: Attempting connection...
00:00:01.001 WIF: Connecting to AP1 4042 Channel 6 BSSId 82:D7:F4:D3:70:AC in mode 11n as XXXX-XXXX-XXXX-00005-6237...
00:00:02.514 WIF: Checking connection...
00:00:02.514 WIF: Attempting connection...
00:00:03.515 WIF: Checking connection...
00:00:03.516 WIF: Attempting connection...
00:00:04.516 WIF: Checking connection...
00:00:04.516 WIF: Attempting connection...
00:00:05.516 WIF: Checking connection...
00:00:05.516 WIF: Attempting connection...
00:00:06.308 QPC: Reset
00:00:06.520 WIF: Checking connection...
00:00:06.520 WIF: Attempting connection...
00:00:07.525 WIF: Checking connection...
00:00:07.525 WIF: Attempting connection...
00:00:08.282 APP: Boot Count 20
00:00:08.526 WIF: Checking connection...
00:00:08.526 WIF: Connected
00:00:08.778 HTP: Web server active on XXXX-XXXX-XXXX-00005-6237 with IP address 192.168.9.138
00:00:09.330 CFG: Saved to flash at FB, Count 33, Bytes 4096
00:00:09.331 WIF: Sending Gratuitous ARP
00:00:09.333 NTP: Sync time...
00:00:09.613 WIF: DNS resolved '2.pool.ntp.org' (162.159.200.123) in 279 ms
00:00:10.619 NTP: No reply from 162.159.200.123
00:00:10.622 NTP: Sync time...
00:00:10.658 WIF: DNS resolved '2.europe.pool.ntp.org' (88.218.226.91) in 34 ms
00:00:11.663 NTP: No reply from 88.218.226.91
00:00:11.666 MQT: Attempting connection...
00:00:11.760 WIF: DNS resolved 'xxxxxxxxxx.azure-devices.net' (20.44.8.226) in 92 ms
00:00:11.761 MQT: Authenticating with an Azure IoT Hub Token
00:00:11.765 MQT: sha256 dataToSign is 'xxxxxxxxxx.azure-devices.net%2Fdevices%2FXXXXX-XXXXX-XXXX-00005
1707944172'
```

</details>

- The device gets stuck in an infinite TLS handshaking loop within the `int WiFiClientSecure_light::_run_until(unsigned target, bool blocking)` function.
- This issue arises from certain internet providers offering restricted internet packages, limiting access to specific services like YouTube and WhatsApp. In such cases, the program receives the same response `BR_SSL_RECVREC` for the `state`, which is defined as `state = br_ssl_engine_current_state(_eng)`. Subsequently, the program becomes blocked in the loop indefinitely.
- This pull request addresses the issue by introducing a 15-second timeout to the `_run_until` loop.

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [ ] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

### CODE REVIEW

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h [hunk 1]
```diff
@@ -131,6 +131,9 @@ class WiFiClientSecure_light : public WiFiClient {
     }
 
   private:
+    unsigned long lastOutActivity;
+    unsigned long lastInActivity;
+    int _loopTimeout=15;
```
Comment (s-hadinger) [inline lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h:136]: Isn't 14 seconds too long? I haven't seen such issue yet, but 5 seconds seems to me already long enough considering that the whole Tasmota is already blocked. 

Maybe we can also reuse the http timeout, hence we need a setter function to change the timeout

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp [hunk 2]
```diff
@@ -557,11 +557,19 @@ int WiFiClientSecure_light::_run_until(unsigned target, bool blocking) {
...
   }
+  lastInActivity = lastOutActivity = millis();
   for (int no_work = 0; blocking || no_work < 2;) {
     if (blocking) {
       // Only for blocking operations can we afford to yield()
       optimistic_yield(100);
     }
+    
+    unsigned long t = millis();
+    if (t-lastInActivity >= ((int32_t) this->_loopTimeout*1000UL) ){
```
Comment (s-hadinger) [inline lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp:568]: This comparison does not work when timer becomes negative after 27 days. Please use `TimeReached()` instead which does the correct comparison

**Review by s-hadinger — CHANGES_REQUESTED** [review]
Thanks.  I haven't had the problem. Great addition

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp (hunk 2 above)
Comment (PabasaraDilshan) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp:568]: Thanks @s-hadinger , I'll look into that

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h (hunk 1 above)
Comment (PabasaraDilshan) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h:136]: Actually, I just copied this 15 seconds from the Arduino implementation of `WiFiClientSecureBearSSL`, https://github.com/esp8266/Arduino/blob/c96dc26792905dc43280587fcdf5ae5af024b286/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp#L249
I thought,  we need that much time in a case where internet connection is slow. In this period whole Tasmota get blocked and can't access the web server of the tasmota. As we have the timeout increment for the mqtt connection. I thought that not be a much problem. Can't we `yield` by default at this `_run_until` loop?. This is the flow after the fix.

```
11:05:59.563 MQT: Authenticating with an Azure IoT Hub Token
11:05:59.972 MQT: TLS connection error: -1000
11:05:59.975 MQT: Connect failed to xxx-xxxx-xxxx-xxx-xx.azure-devices.net:8883, rc -2. Retry in 80 sec
11:07:20.656 MQT: Attempting connection...
11:07:21.248 MQT: Authenticating with an Azure IoT Hub Token
11:07:21.687 MQT: TLS connection error: -1000
11:07:21.690 MQT: Connect failed to xxx-xx-xxx-xxx-xx.azure-devices.net:8883, rc -2. Retry in 90 sec
11:08:52.488 MQT: Attempting connection...
11:08:52.621 MQT: Authenticating with an Azure IoT Hub Token
11:08:52.927 MQT: TLS connection error: -1000
11:08:52.929 MQT: Connect failed to xxx-xx-xxx-xxx-xx.azure-devices.net:8883, rc -2. Retry in 100 sec
```

Also can you please point out this  `http timeout` method that you mentioned.

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp (hunk 2 above)
Comment (PabasaraDilshan) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp]: As `TimeReached()` function is support function for tasmota flow, I just implemented same comparison in this fix

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h (hunk 1 above)
Comment (PabasaraDilshan) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h]: I set the timeout to 5 seconds and checked with simulating the network connection speed. It worked without having  an issue. @s-hadinger

Comment (s-hadinger) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h]: Thanks, let's start with 5 seconds. We can add it as customizable if needed.

**Review by s-hadinger — APPROVED** [review]

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp [hunk 3]
```diff
@@ -557,17 +557,15 @@ int WiFiClientSecure_light::_run_until(unsigned target, bool blocking) {
...
+  uint32_t t = millis();
   for (int no_work = 0; blocking || no_work < 2;) {
     if (blocking) {
       // Only for blocking operations can we afford to yield()
       optimistic_yield(100);
     }
     
-    unsigned long t = millis();
-    if (t-lastInActivity >= ((int32_t) this->_loopTimeout*1000UL) ){
+    if (((int32_t)(millis()-(t+(uint32_t) this->_loopTimeout*1000UL))>=0)){
```
Comment (eku) [inline lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp:567]: `if (((int32_t)(millis()-(t+this->_loopTimeout))>=0)){`

Remove multiplication from loop.

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h [hunk 4]
```diff
@@ -131,9 +131,7 @@ class WiFiClientSecure_light : public WiFiClient {
     }
 
   private:
-    unsigned long lastOutActivity;
-    unsigned long lastInActivity;
-    int _loopTimeout=15;
+    int _loopTimeout=5;
```
Comment (eku) [inline lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h:134]: `uint32_t _loopTimeout=5000;

**Review by eku — CHANGES_REQUESTED** [review]

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp (hunk 3 above)
Comment (PabasaraDilshan) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp]: Thanks @eku, I made that change

File: lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h (hunk 4 above)
Comment (PabasaraDilshan) [reply lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h]: Done

**Comment by s-hadinger** [comment]
Awesome, thanks

**Pull request merged**